### ISSUE TITLE
Primary nav toggles should use buttons instead of links, with correct ARIA markup

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail.org#307]
The main menu’s `data-open-subnav` elements should be `<button>` elements instead of anchors so they are correctly announced to screen reader users.

They seem to follow a [disclosure](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/) widget pattern, so also shouldn’t use `aria-haspopup`, which is intended for `dialog`, `menu` and other advanced ARIA roles, which our implementation doesn’t use.
